server.py

```diff
@@ -52,6 +52,9 @@
 def login():
     # Get instance from login field
     handle = request.args.get("handle", "alexkalopsia@mastodon.social")
+    # FIXME: this actually doesn't work correctly because my handle may be @alice@server.example 
+    # but the instance URL might be social.server.example, you'd need to do a webfinger request 
+    # to discover their profile-page
```

Comment:
Thank you for the PR, most of the changes are very much in line with how I ended up addressing things in the `dev` branch yesterday. 

It doesn't help that I didn't add any contribution guidelines in the README, underlying how iteration work happens in `dev`, so we did pretty much the same thing in parallel but I failed at communicating it. 

If you do find some spare time, I would very much love some help when it comes to  this specific part, that still feels very obscure to me.